Source/player.cpp

```diff
@@ -3203,7 +3198,7 @@ void RemovePlrMissiles(int pnum)
 	}
 
 	for (auto &missile : Missiles) {
-		if (missile._mitype == MIS_STONE && missile._misource == pnum) {
+		if (missile._mitype == MIS_STONE && &Players[missile._misource] == &player) {

```

Comment:
This made me a little nervous, but tracing the code for Rune of Stone suggests that `MIS_STONE` should indeed always have a player as its source. Note, I wasn't able to test in Hellfire due to an unrelated error.

Reply:
Yeah, I'm hoping we can make a function on missiles that will return the player /monster or null